spec/models/immunisation_import_row_spec.rb

```diff
@@ -925,12 +925,13 @@
       end
     end
 
-    context "with an existing vaccination record" do
+    context "with an existing, discarded vaccination record" do
```

Comment:
I can see the logic of combining these two tests, but we're now testing two things in one test (the matching and the undiscarding). We can use `context`s instead to make it clearer.

If we leave it as it was, we can then introduce a context to handle the discarded case:

```rb
context "and the record has been previously discarded" do
  before { existing_vaccination_record.discard! }

  it "stages changes discarded at to nil" do
    expect(vaccination_record.pending_changes["discarded_at"]).to be_nil
  end
end
```